ci/README.md

```diff
@@ -48,12 +48,58 @@ following:
 
 ## `check_enterprise_merge.sh`
 
-When you open a PR on community, if it creates a conflict with
-enterprise-master, the check-merge-to-enterprise will fail. Say your branch name
-is `$PR_BRANCH`, we will refer to `$PR_BRANCH` on community as
-`community/$PR_BRANCH` and on enterprise as `enterprise/$PR_BRANCH`. If the
-job already passes, you are done, nothing further required! Otherwise follow the
-below steps. First make sure these two things are the case:
+This check exists to make sure that we can always merge the `master` branch of
+`community` into the `enterprise-master` branch of the `enterprise` repo.
+There are two conditions in which this check passes:
+1. There are no merge conflicts between your PR branch and `enterprise-master`.
+2. There are merge conflicts, but there is a branch with the same name in the
+   enterprise repo that:
+   1. Contains the last commit of the community branch with the same name.
+   2. Merges cleanly into `enterprise-master`
+
+If the job already passes, you are done, nothing further required! Otherwise
+follow the below steps.
+
+### Prerequisites
+
+Before continuing with the real steps make sure you have done the following
+(this only needs to be done once):
+1. You have enabled `git rerere` in globally or in your enterprise repo
+   ([docs](https://git-scm.com/docs/git-rerere), [very useful blog](https://medium.com/@porteneuve/fix-conflicts-only-once-with-git-rerere-7d116b2cec67#.3vui844dt)):
+   ```bash
+   # Enables it globally for all repos
+   git config --global rerere.enabled true
+   # Enables it only for the enterprise repo
+   cd <enterprise-repo>
+   git config rerere.enabled true
+   ```
+2. You have set up the `community` remote on your enterprise as
+   [described in CONTRIBUTING.md](https://github.com/citusdata/citus-enterprise/blob/enterprise-master/CONTRIBUTING.md#merging-community-changes-onto-enterprise).
+
+
+#### Important notes on `git rerere`
+
+This is very useful as it will make sure git will automatically redo merges that
+you have done before. However, this has a downside too. It will also redo merges
+that you did, but that were incorrect. Two work around this you can use these
```

Comment:
What is an incorrect merge?

Reply:
i.e. merges that you did, but later you found out you merged it in an incorrect way, e.g. you kept both chunks, but should have removed one.